pkg/controllermanager/bootstrappers/bootstrap.go

```diff
@@ -7,23 +7,61 @@ package bootstrappers
 import (
 	"context"
 	"fmt"
+	"strconv"
 	"time"
 
 	"github.com/go-logr/logr"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/client-go/discovery"
 	kubernetesclientset "k8s.io/client-go/kubernetes"
 	"k8s.io/client-go/rest"
 	"k8s.io/utils/clock"
+	"k8s.io/utils/ptr"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	gardencorev1beta1 "github.com/gardener/gardener/pkg/apis/core/v1beta1"
 	v1beta1constants "github.com/gardener/gardener/pkg/apis/core/v1beta1/constants"
+	gardenerutils "github.com/gardener/gardener/pkg/utils/gardener"
 	secretsutils "github.com/gardener/gardener/pkg/utils/secrets"
 	secretsmanager "github.com/gardener/gardener/pkg/utils/secrets/manager"
 	"github.com/gardener/gardener/pkg/utils/version"
 )
 
+const (
+	// ConfigMapsPerShootAnnotation is the annotation key for the amount of ConfigMaps created by Gardener in the project namespace.
+	ConfigMapsPerShootAnnotation = "gardener.cloud/configmaps-per-shoot"
+	// SecretsPerShootAnnotation is the annotation key for the amount of Secrets created by Gardener in the project namespace.
+	SecretsPerShootAnnotation = "gardener.cloud/secrets-per-shoot"
+	shootCountResource        = "count/shoots.core.gardener.cloud"
+)
+
+// ResourceQuotaUsages describes the resource quota usages per shoot cluster in project namespaces.
+type ResourceQuotaUsages struct {
+	Annotation       string
+	QuotaKey         corev1.ResourceName
+	ExpectedPerShoot int
+}
+
+var (
+	// PerShootQuotaDescriptors describes resources that Gardener creates per Shoot Cluster in the project namespace.
+	// Exposed for testing.
+	PerShootQuotaDescriptors = []ResourceQuotaUsages{
+		{
+			ConfigMapsPerShootAnnotation,
+			"count/configmaps",
+			len(gardenerutils.GetShootProjectConfigMapSuffixes()),
+		},
+		{
+			SecretsPerShootAnnotation,
+			"count/secrets",
+			len(gardenerutils.GetShootProjectSecretSuffixes()),
+		},
+	}
+)
```

Comment:
Similar to above, I'm questioning whether these need to be exported since they're only used in this file (and associated tests).
```suggestion
type ResourceQuotaUsages struct {
	Annotation       string
	QuotaKey         corev1.ResourceName
	ExpectedPerShoot int
}

var (
	// PerShootQuotaDescriptors describes resources that Gardener creates per Shoot Cluster in the project namespace.
	// Exposed for testing.
	PerShootQuotaDescriptors = []ResourceQuotaUsages{
		{
			ConfigMapsPerShootAnnotation,
			"count/configmaps",
			len(gardenerutils.GetShootProjectConfigMapSuffixes()),
		},
		{
			SecretsPerShootAnnotation,
			"count/secrets",
			len(gardenerutils.GetShootProjectSecretSuffixes()),
		},
	}
)
```

Reply:
For this one, I would like to keep it exported, as this slice is not used as a parameter for the bumper, but as a kind of "const". I do not want to pass it to every function that needs it, but like to keep it in this pattern of overwriting it for the test case.